src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -2053,23 +2061,31 @@ public unsafe void HandleEvents(Interop.Sys.SocketEvents events)
                 events |= Interop.Sys.SocketEvents.Read | Interop.Sys.SocketEvents.Write;
             }
 
-            AsyncOperation? receiveOperation =
+            ReadOperation? receiveOperation =
                 (events & Interop.Sys.SocketEvents.Read) != 0 ? _receiveQueue.ProcessSyncEventOrGetAsyncEvent(this) : null;
-            AsyncOperation? sendOperation =
+            WriteOperation? sendOperation =
                 (events & Interop.Sys.SocketEvents.Write) != 0 ? _sendQueue.ProcessSyncEventOrGetAsyncEvent(this) : null;
 
             // This method is called from a thread pool thread. When we have only one operation to process, process it
             // synchronously to avoid an extra thread pool work item. When we have two operations to process, processing both
             // synchronously may delay the second operation, so schedule one onto the thread pool and process the other
             // synchronously. There might be better ways of doing this.
-            if (sendOperation == null)
+            if (sendOperation is null)
+            {
+                if (receiveOperation is not null)
+                {
+                    _receiveQueue.ProcessAsyncOperation(receiveOperation);
+                }
+            }
+            else if (receiveOperation is null)
             {
-                receiveOperation?.Process();
+                _sendQueue.ProcessAsyncOperation(sendOperation);
             }
             else
             {
-                receiveOperation?.Schedule();
-                sendOperation.Process();
+                // When there are two operations perfer to run the receive operation and scehedule the send.
+                sendOperation.Schedule();
+                _receiveQueue.ProcessAsyncOperation(receiveOperation);
```

Comment:
It seems like these few lines are the whole purpose of the PR.  What's the need for the rest of the changes?

Reply:
To compile?

Can't use `public AsyncOperation Next` on the base class as that's not the same type as `TOperation` so need to move it down

Reply:
I'm not following, so maybe I'm misunderstanding.  Isn't what you're trying to accomplish the same as:
```C#
if (sendOperation is null)
{
    receiveOperation?.Process();
}
else if (receiveOperation is null)
{
    sendOperation.Process();
}
else
{
    sendOperation.Schedule();
    receiveOperation.Process();
}
```
?

Reply:
No, because `.Process()` then casts it to a `IThreadPoolWorkItem`
```csharp
((IThreadPoolWorkItem)this).Execute()
```
To call
```csharp
void IThreadPoolWorkItem.Execute() => AssociatedContext.ProcessAsyncReadOperation(this);
```
To call
```csharp
void ProcessAsyncReadOperation(ReadOperation op) => _receiveQueue.ProcessAsyncOperation(op);
```
Whereas I just want to call 
```csharp
_receiveQueue.ProcessAsyncOperation(op)
```
Directly; as it seems a bit simpler and doesn't need the interface call?

Reply:
Though from the errors it looks more like it should be
```csharp
receiveOperation.AssociatedContext._receiveQueue.ProcessAsyncOperation(receiveOperation);
```
Might be able to tidy that up by turning it back into
```csharp
public void Process() => AssociatedContext._receiveQueue.ProcessAsyncOperation(receiveOperation);
```

Reply:
Ah, can do it with a bit less churn